cl/users/models.py

```diff
@@ -283,10 +284,71 @@ class BackoffEvent(AbstractDateTimeModel):
         help_text="The next retry datetime for exponential backoff events.",
     )
 
+    @property
+    def under_waiting_period(self) -> bool:
+        """Does the backoff event is under waiting period?
+
+        :return bool: True if so, False if not.
+        """
+        if now() < self.next_retry_date:
+            return True
+        else:
+            return False
+
     def __str__(self) -> str:
         return f"Backoff event for {self.email_address}, next: {self.next_retry_date}"
 
 
+class EmailSent(AbstractDateTimeModel):
+    """Stores email messages."""
+
+    user = models.ForeignKey(
+        User,
+        help_text="The user that this message is related to in case of users "
+        "change their email address we can send failed email to the user's "
+        "new email address, this is optional in case we send email to an"
+        "email address that doesn't belong to a CL user.",
+        related_name="emails",
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+    )
+    message_id = models.UUIDField(
+        help_text="Message ID", default=uuid.uuid4, editable=False
+    )
+    from_email = models.CharField(
+        help_text="From email address", max_length=254
+    )
+    to = models.CharField(
+        help_text="Recipient email address",
+        max_length=254,
+        blank=True,
+    )
+    bcc = models.CharField(
+        help_text="BCC email address", max_length=254, blank=True
+    )
+    cc = models.CharField(
+        help_text="CC email address", max_length=254, blank=True
+    )
+    reply_to = models.CharField(
+        help_text="Reply to address", max_length=254, blank=True
+    )
+    subject = models.CharField(help_text="Subject", max_length=989, blank=True)
```

Comment:
Where did 989 come from? Postgres stores charfields the same as textfields internally, so you don't get any performance boost from using a charfield. Unless we're totally certain that 989 is the magic number we can trust forever, I usually err on the side of using textfields when there's any doubt.

Reply:
Changed to TextField,  I read that the max length of an email subject is 998 characters (the 989 was my mistake) according to [rfc2822](http://www.faqs.org/rfcs/rfc2822.html), however, if there are not any performance improvements using a CharField, I think is better to use a TextField to be sure as you described.